telemetry/telemetry.proto

```diff
@@ -4,6 +4,72 @@ import 'nanopb.proto';
 
 option java_outer_classname="TelemetryProto";
 
-message Telemetry {
-  optional int32 timestamp_ms = 1;
+//A DeviceInfoMessage should only be sent if a request is received from the telemetry viewer (ex. if the
+//		viewer sees a device it hasn't registered yet)
+//Otherwise, the viewer will just reference its config files (stored offline) for this information
+//Note: formerly NameIDMessage
+message DeviceInfoMessage { //send the device ID with its corresponding information
+	enum READING_TYPE {
+		VOLTAGE = 1;
+		CURRENT = 2;
+		TEMPERATURE = 3;
+		MCC = 4;
+		DRIVER_CONTROL = 5; //i.e. pedal board
+		OTHER = 127;
+	}
+	optional uint32 id = 1;
+	optional READING_TYPE readingType = 2 [default = OTHER];
+	optional string name = 3 [(nanopb).max_size = 32];
+	optional uint32 expectedSampleDelay = 4; //expected time between sample readings
+	optional uint32 longestAllowedSampleDelay = 5;
+	optional uint32 numAggregateSamples = 6; //= numSamples in StatisticalAggregate (below)
+	optional string units = 7 [(nanopb).max_size = 4]; //max of four letters (mV, A, degC, mph, etc.)
+	optional uint32 minSensorLimit = 8;
+	optional uint32 maxSensorLimit = 9;
+	optional uint32 resolution = 10;
+	optional uint32 accuracy = 11;
```

Comment:
Sorry, could you clarify that a bit? Did you mean having two separate MeasurementAggregate message types (a ReadingAggregate and a DelayAggregate)? I originally thought that both could be sent together as one message since it seemed reasonable to send both at the same interval, but did you have something else in mind?